lib/rubocop/target_ruby.rb

```diff
@@ -192,14 +192,40 @@ def version_from_gemspec_file(file)
         required_ruby_version(processed_source.ast).first
       end
 
+      def version_from_right_hand_side(right_hand_side)
+        if right_hand_side.array_type?
+          version_from_array(right_hand_side)
+        elsif gem_requirement?(right_hand_side)
+          remove_patch_from_version(right_hand_side.children.last.value)
+        else
+          remove_patch_from_version(right_hand_side.value)
+        end
+      end
+
       def version_from_array(array)
-        versions = array.children.map { |v| version_from_str(v.is_a?(String) ? v : v.str_content) }
-        versions.compact.min
+        array.children.map { |v| remove_patch_from_version(v.value) }
       end
 
-      def version_from_str(str)
-        str.match(/^(?:>=|<=)?\s*(?<version>\d+(?:\.\d+)*)/) do |md|
-          md[:version].to_f
+      def remove_patch_from_version(str)
+        str.to_s.match(/^(?<prefix>>=?|<=?|~>|!=)?\s*(?<version>\d+(?:\.\d+)*)/) do |md|
+          "#{md[:prefix]}#{md[:version].to_f}"
+        end
+      end
+
+      # Helper to check if the requirement is satisfied by a given known ruby version
+      # @api private
+      class Matcher
```

Comment:
I think it would be best to do without this class and use the `Gem::Requirement` as a matcher. Note that matchers typically better constructed without the value to match; that value is passed to the matching method (like `satisfied_by?`).